protoarray/src/main/java/tech/pegasys/teku/protoarray/ProtoArrayForkChoiceStrategy.java

```diff
@@ -123,7 +145,11 @@ public void maybePrune(Bytes32 finalizedRoot) {
 
   // Internal
   private static void processBlocksInStoreAtStartup(ReadOnlyStore store, ProtoArray protoArray) {
+    List<Bytes32> alreadyIncludedBlockRoots =
+        protoArray.getNodes().stream().map(ProtoNode::getBlockRoot).collect(Collectors.toList());
+
     store.getBlockRoots().stream()
```

Comment:
We should be able to optimize this more by adding an API like `store.processBlocks(Consumer<SignedBlockAndState> consumer, Set<Bytes32> blockRoots)`, but we can save that for a follow-up.